### DISCUSSION TITLE
authentication to interactive browser session

### DISCUSSION THREAD

**Discussion by yeoldegrove** [Budibase/budibase#9645] [Help] [answered]
I would like to do some automatic screenshots of my budibase app using puppeteer (https://pptr.dev).
I need to authenticate to the app though ... It works like a charm when I use the cookies issued to a "real" browser session (after logging in manually/interactively).
What would be the correct way to authenticate with a puppeteer browser session (besides scripting to login with user/password) or any other interactive browser session that does not have the ability to log in interactively.

The API key mentioned here does not seem to work: https://docs.budibase.com/docs/public-api
It only works for the API, not a browser session.

**Comment by dewdeded** [comment]
I  have the same question.

**Comment by melohagan** [comment, marked as answer]
You could try hitting the authentication endpoint to get what you need for your Budibase session, e.g. the `x-budibase-token` header.

For example:
```
curl -v -d '{"username":"test@budibase.com", "password":"pass"}' -H "Content-Type: application/json" -X POST "http://localhost:10000/api/global/auth/default/login" 
```

**Reply by yeoldegrove** [reply]
@melohagan @dewdeded Sorry for the late reply. Work like a charm. Adding `--cookie-jar -` to the curl call allows me to filter the cookies and save them to a tmp file for authentication with an additional `curl` or puppeteer.
